### ISSUE TITLE
Enhance Breadcrumb Component Responsiveness and Accessibility

### ISSUE THREAD

**Issue by oladetoungee** [coingaming/moon-light#214]
Currently, the number of visible breadcrumbs is static, and ARIA attributes for screen readers are missing.

We can dynamically adjust the number of visible breadcrumbs based on screen size.

For Accessibility Improvements, we can add aria-current="page" to the current breadcrumb and wrap the navigation in a nav element with an aria-label="breadcrumb" for better screen reader support.
